src/interpreter/mod.rs

```diff
@@ -547,13 +598,12 @@ fn eval_var_decl_stmt(
 }
 
 fn eval_expr(
```

Comment:
I am glad you asked! They were originally (long time ago) a part of the interpreter, but a pretty nasty borrowchecker problem surfaced. Unfortunately I now don't remember what it was or even if it's still there 😂

Reply:
Just checked. This is the problem:

```rust
for (line, stmt) in self.prog.stmts()[range].iter().enumerate() {
    self.eval_stmt(line, stmt)?;
    self.pc += 1;
}
```

The for loop creates a shared borrow on `self.prog`, but `self.eval_stmt(line, stmt)` creates a unique borrow on `self`, which conflicts. I had the execution part of the interpreter initially artificially split to separate struct so that this doesn't happen, but it was far less readable and more boilerplatey, so I went for the simple functions instead.